core/API/Proxy.php

```diff
@@ -22,6 +22,8 @@
 // prevent upgrade error eg from Matomo 3.x to Matomo 4.x. Refs https://github.com/matomo-org/matomo/pull/16468
 // the `false` is important otherwise it would fail and try to load the proxy.php file again.
 if (!class_exists('Piwik\API\NoDefaultValue', false)) {
```

Comment:
Looking at this — can we remove it in Matomo 5.x? Would be one less exception to PSR1.

Reply:
It was added to fix a possible issue when upgrading. So removing it might bring the problem back if someone now updates from an old 3.x.